Challenge-1/src/agent.spec.ts

```diff
@@ -0,0 +1,142 @@
+  const OTHER_FUNCTION_ABI =
+    "function otherFunction(uint256 agentId, address, string metadata, uint256[] chainIds)";
+
+  // Setup for the tests
+  beforeAll(() => {
+    handleTransaction = provideTransaction(
+      CREATE_BOT_FUNCTION,
+      UPDATE_BOT_FUNCTION,
+      BOT_DEPLOYED_ADDRESS,
+      BOT_UPDATE_EVENT,
+      NETHERMIND_DEPLOYER_ADDRESS,
+    );
+  });
+
+  // Test suite for the handleTransaction function
+  describe("handleTransaction", () => {
+    const provideInterface = new Interface([
+      CREATE_BOT_FUNCTION,
+      UPDATE_BOT_FUNCTION,
+      OTHER_FUNCTION_ABI,
+    ]);
+
+    // Test for bot creation
+    it("should find created bot", async () => {
+      const tx: TransactionEvent = new TestTransactionEvent()
+        .setTo(BOT_DEPLOYED_ADDRESS)
+        .setFrom(NETHERMIND_DEPLOYER_ADDRESS)
+        .addTraces({
+          function: provideInterface.getFunction("createAgent"),
+          to: mockBotDeployedAddress,
+          from: mockNethermindAddress,
+          arguments: args,
+        });
+
+      // Call the handleTransaction function with the mock transaction event
+      const expectedFinding = await handleTransaction(tx);
+
+      expect(expectedFinding).toEqual([
+        Finding.fromObject({
+          name: "Bot Creation",
+          protocol: "ethereum",
+          description: `Bot created`,
+          alertId: "BOT-1",
+          severity: FindingSeverity.Low,
+          // timestamp: new Date(),
+          // toString: [Function],
+          type: FindingType.Info,
+          metadata: {
+            // address: mockNethermindAddress,
+            botDeployedAddress: mockBotDeployedAddress,
+          },
+          addresses: [],
+          labels: [],
+          uniqueKey: "",
+          source: {},
+        }),
+      ]);
+    });
+
+    // Test for bot update
+    it("should find updated bot", async () => {
+      const tx: TransactionEvent = new TestTransactionEvent()
+        .setTo(BOT_DEPLOYED_ADDRESS)
+        .setFrom(NETHERMIND_DEPLOYER_ADDRESS)
+        .addTraces({
+          function: provideInterface.getFunction("updateAgent"),
+          to: mockBotDeployedAddress,
+          from: mockNethermindAddress,
+          arguments: mockUpdateAgentEventData2,
+        });
+
+      const expectedFinding = await handleTransaction(tx);
+
+      expect(expectedFinding).toEqual([
+        Finding.fromObject({
+          name: "Bot Updating",
+          protocol: "ethereum",
+          description: `Bot updated`,
+          alertId: "BOT-2",
+          severity: FindingSeverity.Low,
+          type: FindingType.Info,
+          metadata: {
+            botDeployedAddress: mockBotDeployedAddress,
+          },
+          addresses: [],
+          labels: [],
+          uniqueKey: "",
+          source: {},
+        }),
+      ]);
+    });
+
+    // Test for no findings for bot creation
+    it("should not have any findings for bot creation", async () => {
```

Comment:
Suggestion for a more clear test description
```suggestion
    it("should not emit findings for function calls other than bot creation and bot update", async () => {
```